// @ps/payscale-design/lib/tooltip/index.js
'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactTooltip = require('react-tooltip');

var _reactTooltip2 = _interopRequireDefault(_reactTooltip);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Icon = function Icon(props) {
    if (props.display === 'icon') {
        return _react2.default.createElement(
            'div',
            { className: 'icon-reverse icon-circle' },
            _react2.default.createElement('i', { className: 'icon-info' })
        );
    } else {
        return null;
    }
};

Icon.propTypes = {
    display: _propTypes2.default.oneOf(['underline', 'icon'])
};

var Tooltip = function Tooltip(props) {
    var className = props.className,
        display = props.display,
        id = props.id,
        position = props.position,
        hoverText = props.hoverText,
        tooltipText = props.tooltipText;

    var underline = display === 'underline' ? 'pxl-tooltip-underline' : '';
    return _react2.default.createElement(
        'span',
        null,
        _react2.default.createElement(
            'span',
            {
                className: 'pxl-tooltip ' + underline + ' ' + className,
                'data-tip': hoverText,
                'data-for': id },
            tooltipText,
            _react2.default.createElement(Icon, { display: display })
        ),
        _react2.default.createElement(_reactTooltip2.default, { id: id, place: position })
    );
};

Tooltip.propTypes = {
    className: _propTypes2.default.string,
    display: _propTypes2.default.oneOf(['underline', 'icon']),
    hoverText: _propTypes2.default.string.isRequired,
    id: _propTypes2.default.string.isRequired,
    position: _propTypes2.default.oneOf(['left', 'top', 'bottom', 'right']),
    tooltipText: _propTypes2.default.string.isRequired
};

Tooltip.defaultProps = {
    className: '',
    display: 'icon',
    position: 'top'
};

module.exports = Tooltip;
