// @ps/payscale-design/lib/privacy-footer/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var PrivacyFooter = function (_React$Component) {
    _inherits(PrivacyFooter, _React$Component);

    function PrivacyFooter() {
        _classCallCheck(this, PrivacyFooter);

        return _possibleConstructorReturn(this, (PrivacyFooter.__proto__ || Object.getPrototypeOf(PrivacyFooter)).apply(this, arguments));
    }

    _createClass(PrivacyFooter, [{
        key: 'render',
        value: function render() {
            var _props = this.props,
                className = _props.className,
                privacyUrl = _props.privacyUrl,
                privacyStatement = _props.privacyStatement,
                termsOfUseUrl = _props.termsOfUseUrl;


            return _react2.default.createElement(
                'footer',
                { className: 'pxl-privacy-footer ' + className },
                _react2.default.createElement(
                    'div',
                    { className: 'container' },
                    _react2.default.createElement(
                        'div',
                        { className: 'row' },
                        _react2.default.createElement(
                            'div',
                            { className: 'col-xs-12' },
                            _react2.default.createElement(
                                'p',
                                { className: 'pxl-footer-privacy-statement' },
                                _react2.default.createElement('span', { className: 'icon-lock' }),
                                '\xA0',
                                privacyStatement
                            )
                        )
                    )
                ),
                _react2.default.createElement(
                    'div',
                    { className: 'container' },
                    _react2.default.createElement(
                        'div',
                        { className: 'col-xs-12' },
                        _react2.default.createElement(
                            'p',
                            null,
                            _react2.default.createElement(
                                'a',
                                { href: termsOfUseUrl },
                                'Terms of Use'
                            ),
                            _react2.default.createElement(
                                'span',
                                null,
                                '\xA0|\xA0'
                            ),
                            _react2.default.createElement(
                                'a',
                                { href: privacyUrl },
                                'Privacy Policy'
                            )
                        )
                    )
                )
            );
        }
    }]);

    return PrivacyFooter;
}(_react2.default.Component);

exports.default = PrivacyFooter;


PrivacyFooter.propTypes = {
    className: _propTypes2.default.string,
    privacyStatement: _propTypes2.default.string.isRequired,
    privacyUrl: _propTypes2.default.string.isRequired,
    termsOfUseUrl: _propTypes2.default.string.isRequired
};

PrivacyFooter.defaultProps = {
    className: ''
};

module.exports = PrivacyFooter;
