// @ps/payscale-design/webpack.config.styles.js
var webpack = require('webpack');
const ExtractTextPlugin = require('extract-text-webpack-plugin');
const OptimizeCssAssetsPlugin = require('optimize-css-assets-webpack-plugin');
var version = require('./package.json').version;

module.exports = {
    entry: {
        'payscale-icons': __dirname + '/node_modules/@ps/payscale-icons/css/payscale-icons-overrides.scss',
        'payscale-icons.min': __dirname + '/node_modules/@ps/payscale-icons/css/payscale-icons-overrides.scss',
        'pxl-grid-flex': __dirname + '/components/styles/pxl-grid-flex.scss',
        'pxl-grid-flex.min': __dirname + '/components/styles/pxl-grid-flex.scss',
        'pxl-grid': __dirname + '/components/styles/pxl-grid.scss',
        'pxl-grid.min': __dirname + '/components/styles/pxl-grid.scss',
        'pxl-base': __dirname + '/components/styles/pxl-base.scss',
        'pxl-base.min': __dirname + '/components/styles/pxl-base.scss',
        'pxl-components': __dirname + '/components/styles/pxl-components.scss',
        'pxl-components.min': __dirname + '/components/styles/pxl-components.scss',
        'pxl-flex': __dirname + '/components/styles/pxl-flex.scss',
        'pxl-flex.min': __dirname + '/components/styles/pxl-flex.scss',
        'pxl': __dirname + '/components/styles/pxl.scss',
        'pxl.min': __dirname + '/components/styles/pxl.scss'
    },
    output: {
        filename: `css/[name].js`,
        path: __dirname + '/dist',
        publicPath: 'https://localhost:8080/dist/'
    },
    plugins: [
        new ExtractTextPlugin(`css/[name].css`),
        new OptimizeCssAssetsPlugin({
            assetNameRegExp: /\.min\.css$/,
            cssProcessor: require('cssnano'),
            cssProcessorOptions: { discardComments: { removeAll: true } }
        })
    ],
    module: {
        rules: [
            { test: /\.s?css$/, use: ExtractTextPlugin.extract(['css-loader', 'sass-loader'])},
            { test: /\.(woff|woff2|eot|ttf|svg)$/, use: `file-loader?name=[name].[ext]&publicPath=https://cdn-payscale.com/resources/pxl/${version}/&outputPath=font/`},
        ]
    },
    devServer: {
        host: 'localhost',
        port: 8080,
        https: true
    }
};
