// @ps/payscale-design/webpack.config.js
var webpack = require('webpack');
var version = require('./package.json').version;

module.exports = {
    entry: {
        'pxl': __dirname + '/components/index.js',
        'pxl.min': __dirname + '/components/index.js'
    },
    output: {
        filename: `js/[name].js`,
        path: __dirname + '/dist',
        publicPath: 'https://localhost:8080/dist/'
    },
    plugins: [
        new webpack.optimize.UglifyJsPlugin({
            include: /\.min\.js$/,
            minimize: true,
            compress: {
                warnings: false
            }
        }),
        new webpack.optimize.ModuleConcatenationPlugin()
    ],
    module: {
        rules: [
            { test: /\.js$/, use: 'babel-loader', exclude: [/node_modules/, /tests/, /lib/, /dist/] },
            { test: /\.(woff|woff2|eot|ttf|svg)$/, use: `file-loader?name=[name].[ext]&publicPath=https://cdn-payscale.com/resources/pxl/${version}/&outputPath=font/`},
        ]
    },
    devServer: {
        host: 'localhost',
        port: 8080,
        https: true
    }
};
