// @ps/payscale-design/lib/options/OtherOption.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.default = OtherOption;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _option = require('../option');

var _option2 = _interopRequireDefault(_option);

var _input = require('../input');

var _input2 = _interopRequireDefault(_input);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function OtherOptionInput(props) {
    var checked = props.checked,
        onChange = props.onChange,
        value = props.value,
        inputProps = props.inputProps;


    if (checked) {
        return _react2.default.createElement(_input2.default, _extends({ onChange: onChange, value: value }, inputProps));
    }
    return null;
}

OtherOptionInput.propTypes = {
    checked: _propTypes2.default.bool,
    inputProps: _propTypes2.default.object,
    onChange: _propTypes2.default.func,
    value: _propTypes2.default.string
};

function OtherOptionControl(props) {
    var enabled = props.enabled,
        checked = props.checked,
        onChange = props.onChange,
        type = props.type;


    if (enabled) {
        return _react2.default.createElement(_option2.default, { type: type, checked: checked, text: 'Other', onChange: onChange });
    }
    return null;
}

OtherOptionControl.propTypes = {
    checked: _propTypes2.default.bool,
    enabled: _propTypes2.default.bool,
    onChange: _propTypes2.default.func,
    type: _propTypes2.default.oneOf(['pills', 'checklist', 'radio'])
};

function OtherOption(props) {
    var checked = props.checked,
        enabled = props.enabled,
        onClickOther = props.onClickOther,
        onChangeOtherInput = props.onChangeOtherInput,
        type = props.type,
        otherInputValue = props.otherInputValue,
        otherInputProps = props.otherInputProps;


    return _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(OtherOptionControl, { type: type, enabled: enabled, checked: checked, onChange: onClickOther }),
        _react2.default.createElement(OtherOptionInput, { checked: checked, onChange: onChangeOtherInput, value: otherInputValue, inputProps: otherInputProps })
    );
}

OtherOption.propTypes = {
    checked: _propTypes2.default.bool,
    enabled: _propTypes2.default.bool,
    onChangeOtherInput: _propTypes2.default.func,
    onClickOther: _propTypes2.default.func,
    otherInputProps: _propTypes2.default.object,
    otherInputValue: _propTypes2.default.string,
    type: _propTypes2.default.oneOf(['pills', 'checklist', 'radio'])
};
