// @ps/payscale-design/lib/input-number/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _input = require('../input');

var _input2 = _interopRequireDefault(_input);

var _NumeralFormat = require('./NumeralFormat');

var _NumeralFormat2 = _interopRequireDefault(_NumeralFormat);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var InputNumber = function (_React$Component) {
    _inherits(InputNumber, _React$Component);

    function InputNumber() {
        _classCallCheck(this, InputNumber);

        return _possibleConstructorReturn(this, (InputNumber.__proto__ || Object.getPrototypeOf(InputNumber)).apply(this, arguments));
    }

    _createClass(InputNumber, [{
        key: 'render',
        value: function render() {
            var _props = this.props,
                value = _props.value,
                formatProps = _props.formatProps,
                formatEnabled = _props.formatEnabled,
                others = _objectWithoutProperties(_props, ['value', 'formatProps', 'formatEnabled']);

            return _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(_input2.default, _extends({}, others, { value: value, pattern: '[0-9]*', type: 'number' })),
                _react2.default.createElement(_NumeralFormat2.default, _extends({ value: value, enabled: formatEnabled }, formatProps))
            );
        }
    }]);

    return InputNumber;
}(_react2.default.Component);

exports.default = InputNumber;


InputNumber.propTypes = {
    className: _propTypes2.default.string,
    formatEnabled: _propTypes2.default.bool,
    formatProps: _propTypes2.default.shape({
        format: _propTypes2.default.func,
        numericFormat: _propTypes2.default.string
    }),
    id: _propTypes2.default.string,
    onChange: _propTypes2.default.func,
    placeholder: _propTypes2.default.string,
    value: _propTypes2.default.any
};

InputNumber.defaultProps = {
    className: '',
    formatEnabled: false,
    formatProps: {}
};

module.exports = InputNumber;
