// @ps/payscale-design/lib/option/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.default = Option;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function OptionIcon(props) {
    var checked = props.checked,
        type = props.type;


    if (type !== 'checklist') {
        if (checked) {
            return _react2.default.createElement('i', { className: 'icon-dot-circled checked' });
        } else {
            return _react2.default.createElement('i', { className: 'icon-circle-thin' });
        }
    } else {
        if (checked) {
            return _react2.default.createElement(
                'div',
                { className: 'icon-reverse icon-square checked' },
                _react2.default.createElement('i', { className: 'icon-check' })
            );
        } else {
            return _react2.default.createElement('i', { className: 'icon-check-empty' });
        }
    }
}

OptionIcon.propTypes = {
    checked: _propTypes2.default.bool,
    type: _propTypes2.default.oneOf(['pills', 'checklist', 'radio'])
};

function Option(props) {
    var id = props.id,
        text = props.text,
        checked = props.checked,
        className = props.className,
        value = props.value,
        onChange = props.onChange,
        type = props.type,
        others = _objectWithoutProperties(props, ['id', 'text', 'checked', 'className', 'value', 'onChange', 'type']);

    return _react2.default.createElement(
        'div',
        { className: className },
        _react2.default.createElement(
            'label',
            { className: 'pxl-option' },
            _react2.default.createElement('input', _extends({
                id: id,
                type: 'checkbox',
                onChange: onChange.bind(null, value),
                value: value,
                checked: checked
            }, others)),
            _react2.default.createElement(
                'div',
                { className: 'pxl-option__icon' },
                _react2.default.createElement(OptionIcon, { type: type, checked: checked })
            ),
            _react2.default.createElement(
                'div',
                { className: 'pxl-option__text' },
                text
            )
        )
    );
}

Option.propTypes = {
    checked: _propTypes2.default.bool,
    className: _propTypes2.default.string,
    id: _propTypes2.default.string,
    name: _propTypes2.default.string,
    onChange: _propTypes2.default.func,
    text: _propTypes2.default.any,
    type: _propTypes2.default.oneOf(['pills', 'checklist', 'radio']),
    value: _propTypes2.default.any
};

Option.defaultProps = {
    type: 'checklist'
};

module.exports = Option;
