// @ps/payscale-design/lib/form-group/index.js
'use strict';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _ErrorMessage = require('./ErrorMessage');

var _ErrorMessage2 = _interopRequireDefault(_ErrorMessage);

var _label = require('../label');

var _label2 = _interopRequireDefault(_label);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var FormGroup = function (_React$Component) {
    _inherits(FormGroup, _React$Component);

    function FormGroup() {
        _classCallCheck(this, FormGroup);

        return _possibleConstructorReturn(this, (FormGroup.__proto__ || Object.getPrototypeOf(FormGroup)).apply(this, arguments));
    }

    _createClass(FormGroup, [{
        key: 'render',
        value: function render() {
            var _props = this.props,
                id = _props.id,
                htmlFor = _props.htmlFor,
                text = _props.text,
                labelClassName = _props.labelClassName,
                children = _props.children,
                error = _props.error,
                className = _props.className,
                errorClassName = _props.errorClassName;


            return _react2.default.createElement(
                'div',
                { id: id, className: error ? 'pxl-form-group pxl-has-error ' + className : 'pxl-form-group ' + className },
                _react2.default.createElement(_label2.default, { htmlFor: htmlFor, text: text, className: labelClassName }),
                children,
                _react2.default.createElement(_ErrorMessage2.default, { error: error, className: errorClassName })
            );
        }
    }]);

    return FormGroup;
}(_react2.default.Component);

FormGroup.propTypes = {
    className: _propTypes2.default.string,
    error: _propTypes2.default.string,
    errorClassName: _propTypes2.default.string,
    htmlFor: _propTypes2.default.string,
    id: _propTypes2.default.string,
    labelClassName: _propTypes2.default.string,
    text: _propTypes2.default.string
};

FormGroup.defaultProps = {
    className: '',
    errorClassName: '',
    id: '',
    labelClassName: ''
};

module.exports = FormGroup;
